x/ecocredit/server/core/proposal_handler.go

```diff
@@ -0,0 +1,29 @@
+package core
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+
+	coretypes "github.com/regen-network/regen-ledger/x/ecocredit/core"
+)
+
+// ProposalKeeper defines the expected interface for ecocredit module proposals.
+type ProposalKeeper interface {
+	NewCreditType(ctx sdk.Context, ctp *coretypes.CreditTypeProposal) error
+}
+
+func NewCreditTypeProposalHandler(k ProposalKeeper) govtypes.Handler {
+	return func(ctx sdk.Context, content govtypes.Content) error {
+		switch c := content.(type) {
+		case *coretypes.CreditTypeProposal:
+			return handleCreditTypeProposal(ctx, k, c)
+		default:
+			return sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "unrecognized software upgrade proposal content type: %T", c)
```

Comment:
Looks like some copy pasta here within the error message.

Reply:
😨 fixed https://github.com/regen-network/regen-ledger/pull/1015/commits/975f386c657b1f2d09e3fd1b561f57727a6a1d10